com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,130 @@
+        if (servicesInProgressState.isEmpty()) {
+            return StepPhase.DONE;
+        }
+
+        getStepLogger().info(Messages.POLLING_IN_PROGRESS_SERVICES);
+
+        Map<String, ServiceOperationType> servicesOperationTypes = getServicesOperationTypes(servicesInProgressState);
+        getStepLogger().debug(Messages.SERVICES_IN_PROGRESS, JsonUtil.toJson(servicesOperationTypes, true));
+        StepsUtil.setTriggeredServiceOperations(execution.getContext(), servicesOperationTypes);
+
+        List<CloudServiceExtended> servicesData = getServicesData(servicesInProgressState);
+        StepsUtil.setServicesData(execution.getContext(), servicesData);
+
+        return StepPhase.POLL;
+    }
+
+    protected List<CloudServiceExtended> getServicesToProcess(ExecutionWrapper execution) {
+        return Collections.singletonList(StepsUtil.getServiceToProcess(execution.getContext()));
+    }
+
+    private List<CloudServiceExtended> getExistingServices(CloudControllerClient controllerClient,
+                                                           List<CloudServiceExtended> servicesToProcess) {
+        return servicesToProcess.parallelStream()
+                                .map(service -> getExistingService(controllerClient, service))
+                                .filter(Objects::nonNull)
+                                .collect(Collectors.toList());
+    }
+
+    private CloudServiceExtended getExistingService(CloudControllerClient controllerClient, CloudServiceExtended service) {
+        CloudService existingService = controllerClient.getService(service.getName(), false);
+        if (existingService != null) {
+            return ImmutableCloudServiceExtended.builder()
+                                                .from(service)
+                                                .metadata(existingService.getMetadata())
+                                                .build();
+        }
+        return null;
+    }
+
+    private Map<CloudServiceExtended, ServiceOperation> getServicesInProgressState(ExecutionWrapper execution,
+                                                                                   List<CloudServiceExtended> existingServices) {
+        Map<CloudServiceExtended, ServiceOperation> servicesOperation = new HashMap<>();
+        for (CloudServiceExtended existingService : existingServices) {
+            ServiceOperation lastServiceOperation = serviceOperationGetter.getLastServiceOperation(execution, existingService);
+            ServiceOperation serviceOperationInProgress = getServiceOperationInProgress(lastServiceOperation);
+            if (serviceOperationInProgress != null) {
+                servicesOperation.put(existingService, serviceOperationInProgress);
+            }
+        }
+        return servicesOperation;
+    }
+
+    private ServiceOperation getServiceOperationInProgress(ServiceOperation lastServiceOperation) {
```

Comment:
You could change this method to:
private boolean isInProgress(ServiceOperation serviceOperation){
  return lastServiceOperation != null && lastServiceOperation.getState() == ServiceOperationState.IN_PROGRESS;
}